src/flint/fmpz.jl

```diff
@@ -164,20 +170,12 @@ end
   return (unsigned(a) >> (Sys.WORD_SIZE - 2) != 1)
 end
 
-function __fmpz_limbs(a::Int)
-  if __fmpz_is_small(a)
-    return Cint(0)
-  end
-  b = unsafe_load(convert(Ptr{Cint}, unsigned(a)<<2), 2)
-  return b
-end
-
 function _hash_integer(a::Int, h::UInt)
-  s::Cint = __fmpz_limbs(a)
-  s == 0 && return Base.hash_integer(a, h)
-  # get the pointer after the first two Cint
-  d = convert(Ptr{Ptr{UInt}}, unsigned(a) << 2) + 2*sizeof(Cint)
-  p = unsafe_load(d)
+  __fmpz_is_small(a) && return Base.hash_integer(a, h)
+  # view it as a BigInt
+  z = unsafe_load(Ptr{BigInt}(unsigned(a) << 2))
```

Comment:
Doesn't this line need a GC.preserve for a?

Reply:
And why not use the new _as_bigint function here?

Reply:
This can't use `_as_bigint` (which BTW is not new, just moved around) because at this point we only have `a` which is a "pointer-disguised-as-Int. 

Note that this code is essentially the same as it was before, I just reorganized it a bit for clarity.